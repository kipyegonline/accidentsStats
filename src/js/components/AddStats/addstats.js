import React, { useState, useRef } from "react";
import $ from "jquery";

import { useSelector } from "react-redux";
import { Redirect, HashRouter as Router } from "react-router-dom";
import PropTypes from "prop-types";
import {
  Button,
  Form,
  FormGroup,
  Label,
  Input,
  FormText,
  Card,
} from "reactstrap";
import Layout from "../UI/Layout";
import AddStatsTable from "./addStatsTable";

const numregex = new RegExp("^[0-9]+$");

export const AddStatitics = ({ auth }) => {
  const [classv, setClassv] = useState(0);
  const [year, setYear] = useState(false);
  const [fatalities, setFatalties] = useState("");
  const [seriousInjuries, setSeriousInjuries] = useState("");
  const [slightInjuries, setSlightInjuries] = useState("");
  const [date, setDate] = useState(new Date().toDateString());
  const [error, setError] = useState("");
  const [success, setSuccess] = useState("");
  const form = useRef(null);
  const btn = useRef(null);

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log("to submit", classv, fatalities);
    //validate
    if (classv === 0) {
      setError("Please select victim class");
    } else if (
      classv > 0 &&
      fatalities > 0 &&
      slightInjuries > 0 &&
      seriousInjuries > 0
    ) {
      //check if numbers were given
      if (
        numregex.test(fatalities) &&
        numregex.test(slightInjuries) &&
        numregex.test(seriousInjuries)
      ) {
        //ALL GOOD
        btn.current.disabled = true;
        setSuccess("Sending");
        setError("");
        //ajax request
        const data = {
          year: year ? new Date().getFullYear() - 1 : new Date().getFullYear(),
          date,
          fatalities,
          seriousInjuries,
          slightInjuries,
          classv,
        };
        console.log(data, "dara");

        $.ajax({
          type: "POST",
          url: "./server/addstats.php?addStats=true",
          data,
          cache: false,
        })
          .then((res) => {
            console.log(res);
            //if successfulll, reset state
            setSuccess(res.status);
            setTimeout(() => {
              setFatalties("");
              setSeriousInjuries("");
              setSlightInjuries("");
              setClassv(0);
              setSuccess("");
              btn.current.disabled = false;
            }, 5000);
          })
          .catch((error) => {
            setSuccess("");
            setError(error.statusText);
            console.error("Error", error, form);
            form.current.reset();
          });
      } else {
        //tell the damn user to enter numbers
        setError("Ensure text fields have number values");
      }
    } else {
      //ensure that everything is there
      setError("All fields are required");
      setTimeout(() => setError(""), 5000);
    }
  };

  return auth.isLoggedIn ? (
    <Layout>
      <AddStatsTable />
      <Form style={formStyle} ref={form} onSubmit={handleSubmit}>
        <FormGroup>
          <p className="text-info text-center">{"Await"}</p>
        </FormGroup>
        <YearBtn sendValue={setYear} year={year} />
        <FormGroup>
          <Label for="selectClass">Select Victim class</Label>
          <ClassSelect getValue={setClassv} selected={classv} data={addClass} />
        </FormGroup>

        <FormGroup>
          <Label for="fatalties">Fatalities</Label>
          <DataInput
            type="text"
            name="Fatalties"
            id="fatalities"
            placeholder=""
            sendValue={setFatalties}
          />
        </FormGroup>
        <FormGroup>
          <Label for="seriousInjuries">Seriously Injured</Label>
          <DataInput
            type="text"
            name="SeriousInjuries"
            id="seriousInjuries"
            placeholder=""
            sendValue={setSeriousInjuries}
          />
        </FormGroup>

        <FormGroup>
          <Label htmlFor="slightInjuries">Slightly Injured</Label>
          <DataInput
            type="text"
            name="SlightInjuries"
            id="slightInjuries"
            placeholder=""
            sendValue={setSlightInjuries}
          />
        </FormGroup>

        <FormGroup>
          <Label htmlFor="Date">Date</Label>
          <DataInput
            type="date"
            name="Date"
            id="date"
            placeholder=""
            sendValue={setDate}
          />
        </FormGroup>
        <p className="text-danger mt-2">{error}</p>
        <p className="text-succcess mt-2">{success}</p>
        <Button className="blue lighten-2 btn-block" ref={btn}>
          Submit
        </Button>
      </Form>
    </Layout>
  ) : (
    <Router>
      <Redirect to={"/login"} />
    </Router>
  );
};
const AddStats = () => {
  const auth = useSelector((state) => state.auth);
  return <AddStatitics auth={auth} />;
};
export const ClassSelect = ({ getValue, selected, data }) => (
  <select
    id="selectClass"
    className="form-control"
    onChange={(e) => getValue(+e.target.value)}
    value={selected}
  >
    <option value={0}>Select Victim class</option>
    {data.map((d) => (
      <option key={d.id} value={d.id}>
        {d.name}
      </option>
    ))}
  </select>
);
ClassSelect.propTypes = {
  getValue: PropTypes.func,
  selected: PropTypes.number,
  data: PropTypes.arrayOf(
    PropTypes.shape({ id: PropTypes.number, name: PropTypes.string })
  ),
};
const formStyle = {
  maxWidth: 600,
  background: "#fff",
  padding: "2rem",
  margin: "1rem auto",
  border: "1px solid yellow",
  zIndex: 10,
};
export const addClass = [
  { id: 1, name: "pedestrians" },
  { id: 2, name: "Drivers" },
  { id: 3, name: "passengers" },
  { id: 4, name: "Pillion pass" },
  { id: 5, name: "pedal Cyclist" },
  { id: 6, name: "Motor cyclists" },
];

export const DataInput = ({ sendValue, type, name, id, placeholder }) => {
  const handleChanges = (e) => {
    if (e.target.type === "submit") return;
    const paramValue = e.target.value;
    if (paramValue.length > 0) {
      e.target.name == "Date" ? sendValue(paramValue) : sendValue(+paramValue);
    } else {
      $(e.target).after(
        "<small class='text-danger mt-1'>This field is required.</small>"
      );
    }
  };

  const handleBlur = (e) => {
    let el = $(e.target);
    if (e.target.type === "submit") return;
    el.next().remove();
    if (e.target.value.length > 0) {
      e.target.style.border = "";
      el.next().remove();
    } else {
      $(e.target).after(
        "<small class='text-danger mt-3'>This field is required.</small>"
      );
      e.target.style.border = "1px solid red";
    }
  };
  return (
    <Input
      type={type}
      name={name}
      id={id}
      className="form-control mb-2"
      placeholder={placeholder}
      onBlur={handleBlur}
      onChange={handleChanges}
    />
  );
};
DataInput.propTypes = {
  sendValue: PropTypes.func,
  type: PropTypes.string,
  name: PropTypes.string,
  id: PropTypes.string,
  placeholder: PropTypes.string,
};
export const YearBtn = ({ sendValue, year }) => {
  return (
    <div className="float-right">
      <button
        type="button"
        className={`${
          year ? "red " : "blue"
        } btn darken-1 btn-md text-white year-btn`}
        onClick={() => sendValue(!year)}
        style={{ fontSize: "1rem" }}
      >
        {year ? "2019" : "2020"}
      </button>
    </div>
  );
};
YearBtn.propTypes = {
  sendValue: PropTypes.func,
  year: PropTypes.bool,
};
export default AddStats;

const hillVince = {
  name: "Vincent Kipago",
  age: 30,
  locale: "Litein",
  occupation: "Web developer",
  hobbies: ["swimming", "Movies", "Football", "documentaries"],
  married: true,
  des: function() {
    return `my name is ${this.name}, (${this.age}) from ${
      this.locale
    } and i am a ${this.occupation}. I am ${this.married ? "" : "not"} married.
     My hobies are ${this.hobbies.join(",")}. 
   `;
  },
};
