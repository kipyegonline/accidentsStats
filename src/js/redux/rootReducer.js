const initState = {
  ntsa: [],
  ntsb: [],
  loadedA: false,
  loadedB: false,
  iclass: [],
  auth: { isLoggedIn: true, userName: "kipyegonline" },
};

function rootReducer(state = initState, action) {
  if (action.type === "ADD_NTSA") {
    return { ...state, ntsa: action.payload, loadedA: true };
  } else if (action.type === "ADD_NTSB") {
    return { ...state, ntsb: action.payload, loadedB: true };
  } else if (action.type === "ADD_CLASS") {
    return { ...state, iclass: action.payload };
  } else {
    return state;
  }
}
export default rootReducer;
