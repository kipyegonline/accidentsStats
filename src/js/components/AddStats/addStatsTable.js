import React from "react";
import { FormGroup } from "reactstrap";
import { Redirect } from "react-router-dom";
import { addClass, DataInput, YearBtn } from "./addstats";
import { Row, Col } from "reactstrap";
import Layout from "../UI/Layout";

const numregex = new RegExp("^[0-9]+$");
function AddStatsTable() {
  const [data, setData] = React.useState([]);
  const [error, setError] = React.useState("");
  const [success, setSuccess] = React.useState("");
  const [year, getYear] = React.useState(false);
  const [date, setDate] = React.useState("");
  const form = React.useRef(null);
  const getValue = (nature, value, name, id) => {
    //first ensure the user has entered date  and year
    date.length > 0
      ? setError("")
      : setError("Enter date and check year button first");
    setTimeout(() => setError(""), 3000);
    //check if incoming data alraedy exist in the data array, also grab its index

    const check = data.find((d) => +d.classv === id);
    const index = data.findIndex((d) => +d.classv === id);
    if (numregex.test(value) && Number(value) > -1) {
      //prepare data
      let payload = {
        date,
        year: year ? new Date().getFullYear() - 1 : new Date().getFullYear(),
      };

      //if it exists, add incoming props
      if (check) {
        // could use spread operator
        if (nature === "fatalities") {
          check.fatalities = +value.trim();
        } else if (nature === "seriousInjuries") {
          check.seriousInjuries = +value.trim();
        } else if (nature === "slightInjuries") {
          check.slightInjuries = +value.trim();
        }
        //finally update the array
        data[index] = check;
      } else {
        //new victim class, add new props (id,name and nature) to paylaod object and its id, then add to the array
        payload[nature] = +value.trim();
        payload["name"] = name;
        payload["classv"] = id;
        payload["injuryName"] = nature;
        //call this hook to add to array
        setData([...data, payload]);
      }
    } else {
      //so, we check whether there is an object in the array, or an object has the prop to be deleted or there is an object or certain index else nullify
      data.length > 0 || data[nature] !== undefined || data[index] !== undefined
        ? delete data[index][nature]
        : null;
      setError("Kindly enter a number");
    }
  };
  const handleSubmit = (e) => {
    e.preventDefault();

    if (data.length < 6) {
      setError("Please fill all the fields in the table");
    } else {
      let sorted = data.sort((a, b) => a.classv - b.classv);
      let missingFields = sorted.some((item) => Object.values(item).length < 8);

      if (!missingFields) {
        sorted.forEach((data) => {
          setError("");
          setSuccess("great!..sending to server...");
          $.ajax({
            type: "POST",
            url: "./server/addstats.php?addStats=true",
            data,
            dataType: "json",
          })
            .then((res) => {
              if (res.status === 200) {
                setSuccess(res.msg);
                setTimeout(() => {
                  setSuccess("");
                  setData([]);
                  getYear(true);
                  form.current.reset();
                }, 5000);
              } else {
                setSuccess("");
                setError(res.msg);
                setTimeout(() => setError(""), 3000);
              }
            })
            .catch((error) => {
              setSuccess("");
              setError(
                "Error submitting.Check internet connection and try again."
              );
              setTimeout(() => setError(""), 4000);
              console.log(error);
            });
        });
      } else {
        setError(
          "Ensure all  fields have numeric values. Kindly check and try again."
        );
      }
    }
  };

  return JSON.parse(localStorage.getItem("accidentStats")) ? (
    <Layout>
      <Row>
        <Col
          className="mt-4 table-responsive"
          sm="12"
          md={{ size: 8, offset: 2 }}
        >
          <form className="form p-3 my-3" ref={form} onSubmit={handleSubmit}>
            <p className="p-2 alert-primary">
              Add accident statitics..Please start with date and year button.
            </p>
            <Row>
              <Col>
                <DataInput
                  type="date"
                  name="Date"
                  id="date"
                  placeholder=""
                  sendValue={setDate}
                />
              </Col>
              <Col>
                <YearBtn sendValue={getYear} year={year} />
              </Col>
            </Row>

            <table className="table table-bordered table-hover table-responsive">
              <thead>
                <tr>
                  <th>
                    <b>Nature</b>
                  </th>
                  {addClass.map((item) => (
                    <th key={item.id}>
                      <b>{item.name}</b>
                    </th>
                  ))}
                </tr>
              </thead>
              <tbody>
                <tr>
                  <td>Fatalties</td>
                  {addClass.map((item) => (
                    <td key={item.id}>
                      <TableInput
                        item={item}
                        nature="fatalities"
                        sendValue={getValue}
                      />
                    </td>
                  ))}
                </tr>
                <tr>
                  <td>Seriously Injured</td>
                  {addClass.map((item) => (
                    <td key={item.id}>
                      <TableInput
                        item={item}
                        nature="seriousInjuries"
                        sendValue={getValue}
                      />
                    </td>
                  ))}
                </tr>
                <tr>
                  <td>Slightly Injured</td>
                  {addClass.map((item) => (
                    <td key={item.id}>
                      <TableInput
                        item={item}
                        nature="slightInjuries"
                        sendValue={getValue}
                      />
                    </td>
                  ))}
                </tr>
              </tbody>
            </table>
            <FormGroup>
              <p className="text-danger">{error}</p>
              <p className="text-success">{success}</p>
            </FormGroup>
            <input
              type="submit"
              value="Add stats"
              className="btn btn-block btn-primary"
            />
          </form>
        </Col>
      </Row>
    </Layout>
  ) : (
    <Redirect to={"/"} />
  );
}
export default AddStatsTable;

const TableInput = ({ item, nature, sendValue }) => {
  const handleChange = (e, id) => {};
  const handleBlur = (e, id) => {
    if (e.target.value.trim().length > 0) {
      e.target.style.border = "none";
    } else {
      e.target.style.border = "1px solid red";
    }

    sendValue(e.target.dataset.nature, e.target.value, e.target.name, id);
  };
  return (
    <input
      type="number"
      name={item.name}
      data-nature={nature}
      minLength={1}
      onChange={(e) => handleChange(e, item.id)}
      onBlur={(e) => handleBlur(e, item.id)}
      className="form-control my-2"
    />
  );
};
