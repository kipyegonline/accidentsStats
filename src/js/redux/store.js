import { createStore } from "redux";
import rootReducer from "./rootReducer";
const devtools = window.__REDUX_DEVTOOLS_EXTENSION__
  ? window.__REDUX_DEVTOOLS_EXTENSION__()
  : (f) => f;
export const store = createStore(rootReducer, devtools);
