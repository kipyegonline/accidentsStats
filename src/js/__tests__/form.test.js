import React from "react";
import Enzyme, { mount, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";
import AddStats, {
  ClassSelect,
  addClass,
  DataInput,
  YearBtn,
  AddStatitics,
} from "../components/AddStats/addstats";

import { MockStore } from "./redux.test";
Enzyme.configure({ adapter: new Adapter() });

//This test will work if the user is aunthenticated
describe("<AddStatsForm/>", () => {
  const wrapper = mount(
    <MockStore>
      <AddStatitics auth={{ isLoggedIn: true, UserName: "Vince" }} />
    </MockStore>
  );

  describe("The form renders the UI", () => {
    test("Matches snapshot", () =>
      expect(toJSON(wrapper.find("form"))).toMatchSnapshot());
    it("has 4 input elements ", () =>
      expect(wrapper.find("input").length).toBe(4));
    //3 not 2 since i also mounted a button from the navbar
    it("has 2 buttons", () => expect(wrapper.find("button")).toHaveLength(4));
  });
  //the paragraphs
  it("has 2 p tags", () => {
    expect(wrapper.find("p").length).toEqual(3);
  });
  it("submits the form unsuccessfully", () => {
    let form = wrapper.find("form");
    form.simulate("submit");
    //set the props as we move on
    wrapper.setProps({ classv: 1 });
    expect(wrapper.find("p.text-danger").text()).toBe(
      "Please select victim class"
    );
  });
});
describe("<Input/> component", () => {
  const _sendValue = jest.fn();
  const wrapper = shallow(
    <DataInput
      sendValue={_sendValue}
      type="text"
      name="SeriousInjuries"
      id="seriousInjuries"
      placeholder=""
    />
  );

  test(" props", () =>
    expect(wrapper.props()).toEqual({
      name: "SeriousInjuries",
      type: "text",
      id: "seriousInjuries",
      className: "form-control",
      placeholder: "",
      onChange: (f) => f,
      onBlur: (f) => f,
    }));

  it("changes on text enter", () => {
    wrapper.simulate("change", { target: { value: "juless" } });
    expect(_sendValue).toBeCalled();
    expect(_sendValue).toBeCalledTimes(1);
    //Because i am ecxpecting a number not a string
    expect(_sendValue).toBeCalledWith(NaN);
  });
  it.skip("blurs", () => {
    wrapper.simulate("blur", { target: { value: "" } });
    expect(wrapper.find("small")).toHaveLength(1);
  });
  // click and change eventevent
});

describe("<ClassSelect/>", () => {
  const _getValue = jest.fn();
  const wrapper = mount(
    <ClassSelect getValue={_getValue} selected="Select test" data={addClass} />
  );
  describe("Render the damn UI", () => {
    //presence of selcet elemente
    test("Presence of aselect element", () =>
      expect(wrapper.find(".form-control")).toHaveLength(1));
    //rendersexact number of  options
    expect(wrapper.find("option")).toHaveLength(addClass.length + 1);
    expect(
      wrapper
        .find("option")
        .at(1)
        .text()
    ).toBe(addClass[0].name);
    //the second  option has correct value and text
    expect(
      wrapper
        .find("option")
        .at(1)
        .prop("value")
    ).toBe(addClass[0].id);
    //check the default option value
    expect(
      wrapper
        .find("option")
        .at(0)
        .text()
    ).toBe("Select Victim class");
  });
  //change event
  it("Changes on select", () => {
    wrapper
      .find("option")
      .at(1)
      .simulate("change");
    expect(_getValue).toBeCalled();
    expect(_getValue).toBeCalledTimes(1);
    expect(_getValue).toBeCalledWith(1);
  });
});
//Button
describe("<YearBtn/>", () => {
  const _sendValue = jest.fn();
  const wrapper = shallow(<YearBtn year sendValue={_sendValue} />);

  it("The year button does the right thing", () => {
    //text
    expect(wrapper.find(".year-btn").text()).toBe("2019");
    //props
    /** 
    expect(wrapper.find(".year-btn").props()).toEqual({
      type: "button",
      children: "2019",
      className: "red  btn darken-1 btn-md text-white year-btn",
      onClick: (f) => f,
      style: { fontSize: "1rem" },
    }); */
  });
  test("click event", () => {
    //click event
    let ybtn = wrapper.find(".year-btn");
    ybtn.simulate("click");
    wrapper.setProps({ year: false });
    expect(_sendValue).toBeCalled();
    expect(_sendValue).toBeCalledTimes(1);
    expect(wrapper.find(".year-btn").text()).toBe("2020");
  });
});
