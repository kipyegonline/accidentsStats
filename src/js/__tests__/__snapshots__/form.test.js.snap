// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddStatsForm/> Tje form renders the UI Matches snapshot 1`] = `
<form
  className=""
  onSubmit={[Function]}
  style={
    Object {
      "background": "#fff",
      "border": "1px solid yellow",
      "margin": "1rem auto",
      "maxWidth": 600,
      "padding": "2rem",
      "zIndex": 10,
    }
  }
>
  <FormGroup
    tag="div"
  >
    <div
      className="form-group"
    >
      <p
        className="text-info text-center"
      >
        Await
      </p>
    </div>
  </FormGroup>
  <YearBtn
    sendValue={[Function]}
    year={false}
  >
    <div
      className="float-right"
    >
      <button
        className="blue btn darken-1 btn-md text-white year-btn"
        onClick={[Function]}
        style={
          Object {
            "fontSize": "1rem",
          }
        }
        type="button"
      >
        2020
      </button>
    </div>
  </YearBtn>
  <FormGroup
    tag="div"
  >
    <div
      className="form-group"
    >
      <Label
        for="selectClass"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <label
          className=""
          htmlFor="selectClass"
        >
          Select Victim class
        </label>
      </Label>
      <ClassSelect
        data={
          Array [
            Object {
              "id": 1,
              "name": "pedestrians",
            },
            Object {
              "id": 2,
              "name": "Drivers",
            },
            Object {
              "id": 3,
              "name": "passengers",
            },
            Object {
              "id": 4,
              "name": "Pillion pass",
            },
            Object {
              "id": 5,
              "name": "pedal Cyclist",
            },
            Object {
              "id": 6,
              "name": "Motor cyclists",
            },
          ]
        }
        getValue={[Function]}
        selected={0}
      >
        <select
          className="form-control"
          id="selectClass"
          onChange={[Function]}
          value={0}
        >
          <option
            value={0}
          >
            Select Victim class
          </option>
          <option
            key="1"
            value={1}
          >
            pedestrians
          </option>
          <option
            key="2"
            value={2}
          >
            Drivers
          </option>
          <option
            key="3"
            value={3}
          >
            passengers
          </option>
          <option
            key="4"
            value={4}
          >
            Pillion pass
          </option>
          <option
            key="5"
            value={5}
          >
            pedal Cyclist
          </option>
          <option
            key="6"
            value={6}
          >
            Motor cyclists
          </option>
        </select>
      </ClassSelect>
    </div>
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <div
      className="form-group"
    >
      <Label
        for="fatalties"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <label
          className=""
          htmlFor="fatalties"
        >
          Fatalities
        </label>
      </Label>
      <DataInput
        id="fatalities"
        name="Fatalties"
        placeholder=""
        sendValue={[Function]}
        type="text"
      >
        <Input
          className="form-control"
          id="fatalities"
          name="Fatalties"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          type="text"
        >
          <input
            className="form-control form-control"
            id="fatalities"
            name="Fatalties"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="text"
          />
        </Input>
      </DataInput>
    </div>
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <div
      className="form-group"
    >
      <Label
        for="seriousInjuries"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <label
          className=""
          htmlFor="seriousInjuries"
        >
          Seriously Injured
        </label>
      </Label>
      <DataInput
        id="seriousInjuries"
        name="SeriousInjuries"
        placeholder=""
        sendValue={[Function]}
        type="text"
      >
        <Input
          className="form-control"
          id="seriousInjuries"
          name="SeriousInjuries"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          type="text"
        >
          <input
            className="form-control form-control"
            id="seriousInjuries"
            name="SeriousInjuries"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="text"
          />
        </Input>
      </DataInput>
    </div>
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <div
      className="form-group"
    >
      <Label
        htmlFor="slightInjuries"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <label
          className=""
          htmlFor="slightInjuries"
        >
          Slightly Injured
        </label>
      </Label>
      <DataInput
        id="slightInjuries"
        name="SlightInjuries"
        placeholder=""
        sendValue={[Function]}
        type="text"
      >
        <Input
          className="form-control"
          id="slightInjuries"
          name="SlightInjuries"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          type="text"
        >
          <input
            className="form-control form-control"
            id="slightInjuries"
            name="SlightInjuries"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="text"
          />
        </Input>
      </DataInput>
    </div>
  </FormGroup>
  <FormGroup
    tag="div"
  >
    <div
      className="form-group"
    >
      <Label
        htmlFor="Date"
        tag="label"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
      >
        <label
          className=""
          htmlFor="Date"
        >
          Date
        </label>
      </Label>
      <DataInput
        id="date"
        name="Date"
        placeholder=""
        sendValue={[Function]}
        type="date"
      >
        <Input
          className="form-control"
          id="date"
          name="Date"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder=""
          type="date"
        >
          <input
            className="form-control form-control"
            id="date"
            name="Date"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder=""
            type="date"
          />
        </Input>
      </DataInput>
    </div>
  </FormGroup>
  <Button
    className="blue lighten-2 btn-block"
    color="secondary"
    tag="button"
  >
    <button
      aria-label={null}
      className="blue lighten-2 btn-block btn btn-secondary"
      onClick={[Function]}
    >
      Submit
    </button>
  </Button>
  <p
    className="text-danger mt-2"
  />
</form>
`;
